proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,199 @@
+# Test Plan
+There are three test categories we are concerned with:
+
+1. DXIL Intrinsic tests: Go through the operators that support vectors and make a list of the DXIL ops used.
+    - The table to reference for this is in HLOperationLower.cpp.
+    - Many, if not all, of these probably have existing test cases that just need to be modified to use vectors
+        - I think that the 'implementation' will largely be editing ShaderOpArithTable.xml, which is used to generate TAEF 'data driven' tests. Minor shader code updates and added input/output of vectors
+    - Some tests may need to be duplicated for the vector variant of the test.
+    - The tests cases in this category are any intrinsics in the below table that have a DXIL::OpCode::* in the DXIL OPCode colum
+
+2. LLVM Native operation tests
+    - TBD: Where do we test these? Tex mentioned some existing tests but it sounds like we don't want to use those. And they aren't HLK tests anyways.
+    - TODO: Discuss how we want to test ops that don't boil down to a single DXIL op. Still seems like something we should test in the HLK?
+
+3. Standard loading and storing of long vectors
+    - TODO: Do we have tests that do this today?
+
+# Vector Sizes to test
+
+I don't think there are any particularily interesting vector sizes to test. So I propose testing sizes of 5, 25, 100, 500, 
+
+# Phases
+Do the test work in two simple phases.
+
+1. Implement and validate (locally against WARP) the 3 test categories.
+2. HLK related work:
+    - Add a SM 6.9 HLK requirement. Includes updating the HLK requirements doc.
+    - Update mm_annotate_shader_op_arith_table.py to annotate the new test cases with HLK GUIDS and requirements
+    - Add new tests to HLK playlist
```

Comment:
Do we understand all the steps required to get these tests ingested into whatever it is that the IHVs receive?

Reply:
I'm familiar with the HLK, but I do need to check in on the latest processes with CD. I suspect a few details have changed.